tests/api_test.py

```diff
@@ -468,6 +471,7 @@ def test_jit_donate_argnames_kwargs_static_argnums(self):
       ("argnums", "donate_argnums", 0),
       ("argnames", "donate_argnames", 'x'),
   )
+  @jtu.run_on_devices("cpu", "gpu", "tpu")
```

Comment:
Please add comments as to why you are doing this.

Reply:
It might make sense to add a constant named
jtu.DEVICES_SUPPORTING_DONATION
or something like that, which serves both purposes.

Reply:
I've added a `jtu.device_supports_buffer_donation` decorator that is based on the values in `mlir._platforms_with_donation`. This will mean that as that list changes, these tests should automatically pick it up.